packages/react-native-reanimated/Common/cpp/worklets/SharedItems/Shareables.cpp

```diff
@@ -46,7 +55,12 @@ jsi::Value makeShareableClone(
   std::shared_ptr<Shareable> shareable;
   if (value.isObject()) {
     auto object = value.asObject(rt);
-    if (!object.getProperty(rt, "__workletHash").isUndefined()) {
+
+    jsi::PropNameID prop = workletCodePropName(rt);
+    if (object.hasProperty(rt, prop)) {
+      jsi::Value code = object.getProperty(rt, prop);
+      shareable = std::make_shared<ShareableString>(code.asString(rt).utf8(rt));
```

Comment:
Returns directly a sharable primitive instead of object which is directly written into `initData` obj